docs/modules/development/pages/build-from-source.adoc

```diff
@@ -0,0 +1,134 @@
+[[build-from-source]]
+= Build from source
+
+Building OpenNMS from source code highly recommends running a *nix system.
+You do not need to run it physically; a virtual machine is sufficient.
+In our community people use the following environments developing our software:
+
+* link:https://en.wikipedia.org/wiki/List_of_Linux_distributions#Debian-based[Debian based] operating system
+* link:https://en.wikipedia.org/wiki/List_of_Linux_distributions#RPM-based[RPM based] operating system
+* Mac OS X
+
+This section describes how to build OpenNMS {page-component-title} with an Ubuntu Desktop system.
+With following these instructions you will have accomplished:
+
+* The source code and a build environment
+* A compiled and assembled version from the latest develop branch
+* Running an OpenNMS {page-component-title} core server instance from a development directory locally
+
+You should be familiar with deploying OpenNMS {page-component-title}.
+Especially setting up package repositories, configuring and initialising the PostgreSQL database.
+Basic knowledge with git is highly recommended.
+
+== Before you begin
+
+Running {page-component-title} from source requires to install the following components on your system:
+
+* OpenJDK 11 development kit need to be installed, verify the correct `javac` and `java` executables are in your search path
+* Install the xref:deployment:core/getting-started.adoc#install-core-instance[OpenNMS package repositories]
+* Install JICMP and JICMP6 packages which allow sending ICMP messages with Java
+* Install link:https://git-scm.com/[git-scm] to check out the source code from the link:https://github.com/OpenNMS/opennms[OpenNMS GitHub repository]
+* Install and initialize xref:deployment:core/getting-started.adoc#setup-postgresql[PostgreSQL server]
```

Comment:
I wonder if you want to mention the alternative: run postgres from docker. I find that is much easier for development:
- no installation needed
- I can use different postgtres versions 
- when the database is corrupt I just restart the container.
Example: sudo docker run -p 5432:5432 postgres:11.3